src/EFCore.PG.NodaTime/Storage/Internal/NpgsqlNodaTimeTypeMappingSourcePlugin.cs

```diff
@@ -0,0 +1,199 @@
+        [NotNull] readonly NpgsqlRangeTypeMapping _timestamptzZonedDateTimeRange;
+        [NotNull] readonly NpgsqlRangeTypeMapping _timestamptzOffsetDateTimeRange;
+        [NotNull] readonly NpgsqlRangeTypeMapping _dateRange;
+
+        #endregion
+
+        /// <summary>
+        /// Constructs an instance of the <see cref="NpgsqlNodaTimeTypeMappingSourcePlugin"/> class.
+        /// </summary>
+        public NpgsqlNodaTimeTypeMappingSourcePlugin()
+        {
+            _timestampLocalDateTimeRange = new NpgsqlRangeTypeMapping("tsrange", typeof(NpgsqlRange<LocalDateTime>), _timestampLocalDateTime);
+            _timestampInstantRange = new NpgsqlRangeTypeMapping("tsrange", typeof(NpgsqlRange<Instant>), _timestampInstant);
+            _timestamptzInstantRange = new NpgsqlRangeTypeMapping("tstzrange", typeof(NpgsqlRange<Instant>), _timestamptzInstant);
+            _timestamptzZonedDateTimeRange = new NpgsqlRangeTypeMapping("tstzrange", typeof(NpgsqlRange<ZonedDateTime>), _timestamptzZonedDateTime);
+            _timestamptzOffsetDateTimeRange = new NpgsqlRangeTypeMapping("tstzrange", typeof(NpgsqlRange<OffsetDateTime>), _timestamptzOffsetDateTime);
+            _dateRange = new NpgsqlRangeTypeMapping("daterange", typeof(NpgsqlRange<LocalDate>), _date);
+
+            var storeTypeMappings = new Dictionary<string, RelationalTypeMapping[]>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "timestamp", new RelationalTypeMapping[] { _timestampInstant, _timestampLocalDateTime } },
+                { "timestamp without time zone", new RelationalTypeMapping[] { _timestampInstant, _timestampLocalDateTime } },
+                { "timestamptz", new RelationalTypeMapping[] { _timestamptzInstant, _timestamptzZonedDateTime, _timestamptzOffsetDateTime } },
+                { "timestamp with time zone", new RelationalTypeMapping[] { _timestamptzInstant, _timestamptzZonedDateTime, _timestamptzOffsetDateTime } },
+                { "date", new RelationalTypeMapping[] { _date } },
+                { "time", new RelationalTypeMapping[] { _time } },
+                { "time without time zone", new RelationalTypeMapping[] { _time } },
+                { "timetz", new RelationalTypeMapping[] { _timetz } },
+                { "time with time zone", new RelationalTypeMapping[] { _timetz } },
+                { "interval", new RelationalTypeMapping[] { _period } },
+
+                { "tsrange", new RelationalTypeMapping[] { _timestampInstantRange, _timestampLocalDateTimeRange } },
+                { "tstzrange", new RelationalTypeMapping[] { _timestamptzInstantRange, _timestamptzZonedDateTimeRange, _timestamptzOffsetDateTimeRange } },
+                { "daterange", new RelationalTypeMapping[] { _dateRange} }
+            };
+
+            var clrTypeMappings = new Dictionary<Type, RelationalTypeMapping>
+            {
+                { typeof(Instant), _timestampInstant },
```

Comment:
Maybe as above with `[typeof(KeyType)] = ...`, but again, either way works for me.

Reply:
Leaving as-is.